File: api-tester/src/defaults/java/com/revenuecat/apitester/java/RedeemWebResultListenerAPI.java
package com.revenuecat.apitester.java;

import androidx.annotation.OptIn;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI;
import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.interfaces.RedeemWebResultListener;

@OptIn(markerClass = ExperimentalPreviewRevenueCatPurchasesAPI.class)
@SuppressWarnings({"unused"})
final class RedeemWebResultListenerAPI {
    static void checkListener(RedeemWebResultListener listener,
                              RedeemWebResultListener.Result result) {
        listener.handleResult(result);
    }

    static void checkRedeemResult(RedeemWebResultListener.Result result) {
        if (result instanceof RedeemWebResultListener.Result.Success) {
            CustomerInfo customerInfo = ((RedeemWebResultListener.Result.Success) result).getCustomerInfo();
        } else if (result instanceof RedeemWebResultListener.Result.Error) {
            PurchasesError error = ((RedeemWebResultListener.Result.Error) result).getError();
        }

        boolean isSuccess = result.isSuccess();
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/RedeemWebResultListenerAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.interfaces.RedeemWebResultListener

@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
@Suppress("unused", "UNUSED_VARIABLE")
private class RedeemWebResultListenerAPI {
    fun checkListener(
        redeemWebPurchaseListener: RedeemWebResultListener,
        result: RedeemWebResultListener.Result,
    ) {
        redeemWebPurchaseListener.handleResult(result)
    }

    fun checkResult(result: RedeemWebResultListener.Result): Boolean {
        val isSuccess: Boolean = result.isSuccess

        when (result) {
            is RedeemWebResultListener.Result.Success -> {
                val customerInfo: CustomerInfo = result.customerInfo
                return true
            }
            is RedeemWebResultListener.Result.Error -> {
                val error: PurchasesError = result.error
                return false
            }
        }
    }
}
